core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/config/update/ConfigUpdateService.java

```diff
@@ -0,0 +1,88 @@
+import com.sequenceiq.cloudbreak.common.service.TransactionService.TransactionRuntimeExecutionException;
+import com.sequenceiq.cloudbreak.core.bootstrap.service.host.ClusterHostServiceRunner;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.provision.ClusterCreationService;
+import com.sequenceiq.cloudbreak.core.flow2.stack.CloudbreakFlowMessageService;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.view.StackView;
+import com.sequenceiq.cloudbreak.service.StackUpdater;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterService;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import java.util.List;
+import javax.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ConfigUpdateService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigUpdateService.class);
+
+    @Inject
+    private StackUpdater stackUpdater;
+
+    @Inject
+    private CloudbreakFlowMessageService flowMessageService;
+
+    @Inject
+    private ClusterService clusterService;
+
+    @Inject
+    private TransactionService transactionService;
+
+    @Inject
+    private ClusterCreationService clusterCreationService;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private ClusterHostServiceRunner clusterHostServiceRunner;
+
+    public void doConfigUpdate(Long stackId) {
+        stackUpdater.updateStackStatus(stackId, DetailedStackStatus.BOOTSTRAPPING_MACHINES);
+        flowMessageService
+            .fireEventAndLog(stackId, UPDATE_IN_PROGRESS.name(), CLUSTER_CONFIG_UPDATE_STARTED);
+        Stack stack = stackService.getByIdWithClusterInTransaction(stackId);
+        clusterHostServiceRunner.updateClusterConfigs(stack, stack.getCluster(), List.of());
+    }
+
+    public void configUpdateFinished(StackView stackView) {
+        try {
+            transactionService.required(() -> {
+                clusterService.updateClusterStatusByStackId(stackView.getId(), AVAILABLE);
+                stackUpdater.updateStackStatus(stackView.getId(), DetailedStackStatus.AVAILABLE,
+                    "Config update finished.");
+                flowMessageService.fireEventAndLog(stackView.getId(), AVAILABLE.name(),
+                    CLUSTER_CONFIG_UPDATE_FINISHED);
+            });
+        } catch (TransactionExecutionException e) {
+            throw new TransactionRuntimeExecutionException(e);
+        }
+    }
+
+    public void handleConfigUpdateFailure(StackView stackView, Exception exception) {
+        if (stackView.getClusterView() != null) {
+            String errorMessage = clusterCreationService.getErrorMessageFromException(exception);
+            clusterService
+                .updateClusterStatusByStackId(stackView.getId(), UPDATE_FAILED, errorMessage);
+            stackUpdater.updateStackStatus(stackView.getId(), DetailedStackStatus.AVAILABLE);
```

Comment:
if the above was in transaction, shouldn't this be in one?

Reply:
fixed